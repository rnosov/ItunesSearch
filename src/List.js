/**
 * Itunes search list component
 *
 * Copyright Â© Roman Nosov 2016
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';

function List({ data,toggles,onToggle }) {
  if (!onToggle) return (
    <ul className="nav nav-sidebar">            
      {data.map( (r, key) => (
        toggles[key]?<li key={key}>{`${r.artistName} - ${r.trackName||r.collectionName||r.wrapperType}`}</li>:void 0
      ))}
      <li></li>            
    </ul>         
  );
  return(
    <ul className="media-list">
      {data.map( (r, key) => (
        <li key={key} className="media">
          <div className="media-left">
            <img className="media-object" src={r.artworkUrl60} alt="No Artwork" width="60" height="60" />
          </div>
          <div className="media-body">
            <h5 className="media-heading">{`${r.artistName}${r.collectionName?' - ' + r.collectionName:''}`}</h5>
            <span className="tag tag-pill tag-info">{r.wrapperType}</span>{ " " }            
            <span className="tag tag-pill tag-info">{r.collectionType}</span>{ " " }            
            <span className="tag tag-pill tag-info">{r.kind}</span>{ " " }
            {r.trackName}
          </div>
          <div className="media-right">
            <button 
              type="button" 
              onClick={onToggle(key)} 
              className={'btn media-object btn-outline-'+(toggles[key]?'warning':'primary')}
            >
              {toggles[key]?'Unf':'F'}avourite
            </button>            
          </div>
        </li>
      ))}
    </ul>   
  );
}

List.propTypes = { 
  data: React.PropTypes.array.isRequired,
  toggles: React.PropTypes.object.isRequired,
  onToggle: React.PropTypes.func,
};

export default List;