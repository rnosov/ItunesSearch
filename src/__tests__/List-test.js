import React from 'react';
import List from '../List';
import renderer from 'react-test-renderer';

describe('List', () => {

  const data = [
  {
    artistName    : 'x',
    collectionName: 'y',
    wrapperType   : 'u',
    collectionType: 'i',
    kind          : 'o',
  },

  {
    artistName    : 'q',
    collectionName: 'w',
    wrapperType   : 'e',
    collectionType: 'r',
    kind          : 'ty',
  },

  ];

  it('renders complex list', () => {
    const instance = renderer.create(<List data={data} toggles={{}} onToggle={ () => false } />);
    const tree = instance.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('renders simple list', () => {
    const instance = renderer.create(<List data={data} toggles={{0:true}} />);
    const tree = instance.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
