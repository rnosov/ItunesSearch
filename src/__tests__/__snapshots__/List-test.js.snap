exports[`List renders complex list 1`] = `
<ul
  className="media-list">
  <li
    className="media">
    <div
      className="media-left">
      <img
        alt="No Artwork"
        className="media-object"
        height="60"
        src={undefined}
        width="60" />
    </div>
    <div
      className="media-body">
      <h5
        className="media-heading">
        x - y
      </h5>
      <span
        className="tag tag-pill tag-info">
        u
      </span>
       
      <span
        className="tag tag-pill tag-info">
        i
      </span>
       
      <span
        className="tag tag-pill tag-info">
        o
      </span>
       
    </div>
    <div
      className="media-right">
      <button
        className="btn media-object btn-outline-primary"
        onClick={false}
        type="button">
        F
        avourite
      </button>
    </div>
  </li>
  <li
    className="media">
    <div
      className="media-left">
      <img
        alt="No Artwork"
        className="media-object"
        height="60"
        src={undefined}
        width="60" />
    </div>
    <div
      className="media-body">
      <h5
        className="media-heading">
        q - w
      </h5>
      <span
        className="tag tag-pill tag-info">
        e
      </span>
       
      <span
        className="tag tag-pill tag-info">
        r
      </span>
       
      <span
        className="tag tag-pill tag-info">
        ty
      </span>
       
    </div>
    <div
      className="media-right">
      <button
        className="btn media-object btn-outline-primary"
        onClick={false}
        type="button">
        F
        avourite
      </button>
    </div>
  </li>
</ul>
`;

exports[`List renders simple list 1`] = `
<ul
  className="nav nav-sidebar">
  <li>
    x - y
  </li>
  <li />
</ul>
`;
